src/foam/dao/jdbc/MySQLJDBCDAO.java

```diff
@@ -0,0 +1,333 @@
+        sink.put(createFObject(resultSet), null);
+      }
+
+      return sink;
+    } catch (Throwable e) {
+      Logger logger = (Logger) x.get("logger");
+      logger.error(e);
+      return null;
+    } finally {
+      closeAllQuietly(resultSet, stmt);
+    }
+  }
+
+  /**
+   * Create the table in the database and return true if it doesn't already exist otherwise it does nothing and returns false
+   * @param of ClassInfo
+   */
+  @Override
+  public boolean createTable(X x, ClassInfo of) {
+    Connection c = null;
+    IndexedPreparedStatement stmt = null;
+    ResultSet resultSet = null;
+
+    try {
+      c = dataSource_.getConnection();
+      DatabaseMetaData meta = c.getMetaData();
+      resultSet = meta.getTables(null, null, tableName_, new String[]{"TABLE"});
+      if ( resultSet.isBeforeFirst() ) {
+        // found a table, don't create
+        return false;
+      }
+
+      StringBuilder builder =threadLocalBuilder_.get()
+        .append("CREATE TABLE ")
+        .append(tableName_)
+        .append("(")
+        .append(getPrimaryKey().createStatement())
+        .append(" ")
+        .append(getPrimaryKey().getSQLType())
+        .append(" primary key,");
+
+      Iterator i = properties_.iterator();
+      while ( i.hasNext() ) {
+        PropertyInfo prop = (PropertyInfo) i.next();
+
+        // Why you skip the primary key? (Ask Kevin)
+        if ( getPrimaryKey().getName().equals(prop.getName()) )
+          continue;
+
+        builder.append(prop.createStatement())
+          .append(" ")
+          .append(prop.getSQLType()); // TODO: is getSQLType guaranteed to return something?
+
+        if ( i.hasNext() ) {
+          builder.append(",");
+        }
+      }
+      builder.append(")");
+
+      // execute statement
+      stmt = new IndexedPreparedStatement(c.prepareStatement(builder.toString()));
+      stmt.executeUpdate();
+      return true;
+    } catch (Throwable e) {
+      Logger logger = (Logger) x.get("logger");
+      logger.error(e);
+      return false;
+    } finally {
+      closeAllQuietly(resultSet, stmt);
+    }
+  }
+
+  /**
+   * Prepare the formatted column names. Appends column names like: (c1,c2,c3)
+   * @param builder builder to append to
+   */
+  public void buildFormattedColumnNames(FObject obj, StringBuilder builder) {
+    // collect columns list into comma delimited string
+    builder.append("(");
+    Iterator i = properties_.iterator();
+    while ( i.hasNext() ) {
+      PropertyInfo prop = (PropertyInfo) i.next();
+/*       if ( "id".equals(prop.getName()) )
+        continue; */
+
+      builder.append(prop.createStatement());
+      if ( i.hasNext() ) {
+        builder.append(",");
+      }
+    }
+    builder.append(")");
+  }
+
+  /**
+   * Prepare the formatted UPDATE string like :  description='etc', name='etc'
+   * @param builder builder to append to
+   */
+  public void buildUpdateFormattedColumnNames(FObject obj, StringBuilder builder) {
```

Comment:
Is this MySQL specific?

Reply:
Yes this is the part used by the put that is specific to MySQL